gsy_framework/sim_results/electric_blue/all_results.py

```diff
@@ -0,0 +1,103 @@
+from collections import defaultdict
+from typing import Dict
+
+from pendulum import from_format, DateTime, Duration
+
+from gsy_framework.constants_limits import DATE_TIME_FORMAT
+from gsy_framework.enums import AvailableMarketTypes
+from gsy_framework.forward_markets.forward_profile import ALLOWED_MARKET_TYPES
+from gsy_framework.sim_results.electric_blue.aggregate_results import (handle_forward_results,
+                                                                       ForwardDeviceStats,
+                                                                       MARKET_RESOLUTIONS)
+from gsy_framework.sim_results.electric_blue.time_series import ForwardDeviceTimeSeries
+
+
+class ForwardResultsHandler:
+    """Calculate all results for each market slot for forward markets."""
+
+    def __init__(self, slot_length: Duration):
+        self.forward_market_enabled = True
+        self.slot_length = slot_length
+        self.orders: Dict[
+            str, Dict[int, Dict[DateTime, Dict]]] = defaultdict(lambda: defaultdict(dict))
+        self.previous_device_stats = {}
+        self.current_device_stats: Dict[
+            int, Dict[DateTime, Dict[str, Dict]]] = defaultdict(lambda: defaultdict(dict))
+        self.device_time_series: Dict[
+            int, Dict[int, Dict[DateTime, Dict[str, Dict]]]] = defaultdict(
+            lambda: defaultdict(lambda: defaultdict(dict)))
+        self._total_memory_utilization_kb = 0.0
+
+    # pylint: disable=unused-argument
+    def update(self, area_dict: Dict, core_stats: Dict, current_market_slot: str) -> None:
+        """
+        Update the forward market aggregated results with bids_offers_trades of current slot.
+        """
+        self._clear_device_statistics()
+        for area_uuid, area_result in core_stats.items():
+            if "forward_market_stats" not in area_result:
+                return
+
+            current_market_dt = from_format(current_market, DATE_TIME_FORMAT) if (
+                current_market := current_market_slot) else ""
+            for market_type_value, market_stats in area_result["forward_market_stats"].items():
+                market_type = int(market_type_value)
+                self._get_bids_offers_trades(area_uuid, market_type, market_stats)
+                current_results = handle_forward_results(current_market_dt, market_stats)
+                self._update_stats_and_time_series(current_results, market_type)
+        self._update_memory_utilization()
+
+    def update_from_repr(self, area_representation: Dict):
+        """
+        Updates the simulation results using area_representation data that arrive from the d3a-web.
```

Comment:
NIT:
```suggestion
        Updates the simulation results using area_representation data that arrive from the gsy-web.
```
Can you please also replace this in the other occurrences? Thanks a lot